### PR TITLE
More detail on everything as code

### PR DESCRIPTION
**Pull request by ivorc**

### CODE REVIEW

File: patterns/everything-as-code.md [hunk 1]
```diff
@@ -21,7 +21,15 @@ Everything (including [infrastructure](../practices/cloud-services.md)) should b
...
 * Code should be automatically scanned for secrets or other sensitive data using standalone tools like [GitGuardian](https://www.gitguardian.com/) or built in tools in [GitLab](https://docs.gitlab.com/ee/user/application_security/secret_detection/) or [GitHub](https://docs.github.com/en/github/administering-a-repository/about-secret-scanning).
+* Prefer convention over configuration as it leads to simpler code.
+* Prefer well structured and expressive code over extensive documentation to avoid documentation getting out of date.
+* Design the interface prior to the implementation and choose vocabulary to make it coherent. This includes external interfaces (e.g. REST API) and internal interfaces of classes, method signatures etc.
+* Adopt test-first approach to minimise waste and increase cohesion of the code.
+* Follow the clean code practices across the whole software stack, see [Clean Code](https://www.oreilly.com/library/view/clean-code/9780136083238/) and [Clean Architecture](https://www.oreilly.com/library/view/clean-architecture-a/9780134494272/).
 
 ## Examples
```
Comment (andyblundell) [inline patterns/everything-as-code.md:30]: Maybe add test code to the list too?

File: patterns/everything-as-code.md [hunk 2]
```diff
@@ -21,7 +21,15 @@ Everything (including [infrastructure](../practices/cloud-services.md)) should b
   * All code is version controlled.
 * Code changes should be automatically checked for code quality using tools like [SonarQube](https://www.sonarqube.org) (as well as via IDE plugins).
 * Code should be automatically scanned for secrets or other sensitive data using standalone tools like [GitGuardian](https://www.gitguardian.com/) or built in tools in [GitLab](https://docs.gitlab.com/ee/user/application_security/secret_detection/) or [GitHub](https://docs.github.com/en/github/administering-a-repository/about-secret-scanning).
+* Prefer convention over configuration as it leads to simpler code.
+* Prefer well structured and expressive code over extensive documentation to avoid documentation getting out of date.
+* Design the interface prior to the implementation and choose vocabulary to make it coherent. This includes external interfaces (e.g. REST API) and internal interfaces of classes, method signatures etc.
+* Adopt test-first approach to minimise waste and increase cohesion of the code.
```
Comment (andyblundell) [inline patterns/everything-as-code.md:27]: Maybe link off to the test page as that mentions test-first too?

**Review by andyblundell — COMMENTED** [review]
Great stuff 👍

**Review by stefaniuk — APPROVED** [review]
👍  LGTM

**Review by andyblundell — APPROVED** [review]

**Pull request merged**